packages/utils/node-resolver-core/src/NodeResolver.js

```diff
@@ -526,7 +525,9 @@ export default class NodeResolver {
 
       return {
         filePath,
-        query: url.search ? parseQueryString(url.search.slice(1)) : undefined,
+        query: url.search
+          ? Object.fromEntries(new URLSearchParams(url.search.slice(1)))
```

Comment:
Is it a problem that this doesn't handle duplicates the same way?

```
> require('querystring').parse('a=b&a=c')
[Object: null prototype] { a: [ 'b', 'c' ] }

> Object.fromEntries(new URLSearchParams('a=b&a=c'))
{ a: 'c' }
```

Reply:
I forgot about that. Then we should also change
https://github.com/parcel-bundler/parcel/blob/461de1149a0312d4c541d8e5fca977a7f3c58f7e/packages/core/types/index.js#L28
to be `{ [string]: string | Array<string> }` as well